Automattic/pocket-casts-ios
# [Tooling] Add script to spell-check `en.lproj` strings

This is a proof-of-concept to spell-check the strings from `en.lproj` to find typos.

This idea comes from a discussion with @danielebogo in Slack DM

## To test

 - Install `aspell` using `brew install aspell` (note that it _might_ be already installed on your Mac by another brew dependency)
 - ~Run `scripts/spell-check-en-strings.rb`~ Run `bundle exec fastlane spellcheck_strings`

<details><summary> 🖼️ This is the output when run on my machine</summary>

_Old output from initial implementation. See [below comment](https://github.com/Automattic/pocket-casts-ios/pull/2265#issuecomment-2417218201) for the new rendering after I migrated the code to be a lane_
![image](https://github.com/user-attachments/assets/dd536c8e-2556-44a7-af70-549fcbe7fedb)
</details>

## What's Next

This is only a proof of concept / draft for now. As a follow-up after this lands, we probably want to:

 - Fix the current typos, and potentially add more acceptable exceptions to the `KNOWN_WORDS` lists if needed
 - Make this run on CI _(Note: this will require to run on a macOS agent because it uses `plutil`[^1])_

[^1]: We could arguably re-implement `plutil` in Swift for Linux if we wanted to, so we could run that script on a Linux machine instead of Mac; but that's out of scope of this PR / spike

## Checklist
